homework-submission/reactweek1/src/App.js

```diff
@@ -0,0 +1,123 @@
+import React, { Component } from 'react';
+import './App.css';
+
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.changeToDone = this.changeToDone.bind(this);
+    this.descriptionHandler = this.descriptionHandler.bind(this);
+    this.deadLineHandler = this.deadLineHandler.bind(this);
+    this.removeHandler = this.removeHandler.bind(this);
+    this.addHandler = this.addHandler.bind(this);
+    this.state = {
+      list: [
+        {
+          "id": 1,
+          "description": "Get out of bed",
+          "deadline": "2017-09-11",
+          "done": true
+        },
+        {
+          "id": 2,
+          "description": "Brush teeth",
+          "deadline": "2017-09-10",
+          "done": false
+        },
+        {
+          "id": 3,
+          "description": "Eat breakfast",
+          "deadline": "2017-09-09",
+          "done": false
+        },
+      ],
+      newDescription: '',
+      newDeadline: ''
+    }
+  }
+
+  changeToDone = (event) => {
+    const id = event.target.id;
+    const listCopy = this.state.list.slice(0);
+    const itemIndex = listCopy.findIndex(el => { return el.id === Number(id) });
+    listCopy[itemIndex].done = !listCopy[itemIndex].done;
+    this.setState({
+      list: listCopy
+    });
+  }
+  descriptionHandler = (newDescription) => {
+    const state = this.state;
+    state['newDescription'] = newDescription;
+    this.setState(state)
```

Comment:
One comment was lost, sorry.
You are mutating the state in these functions, and you should avoid it at all costs. [here you can find out more why](https://medium.com/pro-react/a-brief-talk-about-immutability-and-react-s-helpers-70919ab8ae7c) you should keep the state immutable.

In this case you can simply do:
```js
descriptionHandler = newDescription => this.setState({ newDescription });
```